students/javarb/sessions/2019/may/Session_34-35-Serialization/Session_34-35-Serialization.md

```diff
@@ -0,0 +1,225 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 34 (21/05/2019 )
+
+- Research homework [on my project](https://github.com/kinbiko/mentoring/projects/7) board.
+
+## Notes
+
+### Homework
+
+#### Part 1
+Figure out the most popular libraries for turning JSON strings into POJOs, and for turning POJOs into JSON.
+
+#### Part 2
+Write a simple project to benchmark their performance for serialization and serialization.
+
+#### Part 3
+Write a blog post (preferably in English) about what you discovered.
+
+I have get to decide:
+
+- how many libraries to check
+- how you can create a benchmark
+- how to write your blog post.
+
+### Solution
+
+#### Part 1
+
+I found this [Top 5 Libraries for Serialization and Deserialization JSON in Java](https://simplesolution.dev/top-5-libraries-for-serialization-and-deserialization-json-in-java/). There they are mentioning following libraries:
+
+- [Gson](https://github.com/google/gson)
+- [Jackson](https://github.com/FasterXML/jackson)
+- [Fastjson](https://github.com/alibaba/fastjson)
+- [Moshi](https://github.com/square/moshi)
+- [Jsoniter](https://github.com/square/moshi)
+
+I found [this other article](http://dgimenes.com/blog/2014/03/14/java-comparing-json-encoding-libraries.html) when the person who wrote is concluding that Gson and Jackson are best libraries but also is mentioning [JSON.org](http://json.org/) library which [seems to be obsolete now](https://stackoverflow.com/a/7623694). There in JSON.org, we can see these list of links to different Java libraries to handle serialization:
+
+- JSON-java.
+- JSONUtil.
+- jsonp.
+- Json-lib.
+- Stringtree.
+- SOJO.
+- json-taglib.
+- Flexjson.
+- Argo.
+- jsonij.
+- fastjson.
+- mjson.
+- jjson.
+- json-simple.
+- json-io.
+- google-gson.
+- FOSS Nova JSON.
+- Corn CONVERTER.
+- Apache johnzon.
+- Genson
+- cookjson.
+- progbase.
+
+Stil is a extensive list it's strange Jackson isn't there.
+
+## Session 34 (21/05/2019 )
+
+#### Part 2 - Benchmark project
+
+I decided to use: **Gson, Jackson and FastJson ** for my checks. In order to use them I needed to include them on `build.gradle` file:
+```java
+dependencies {
+	//...
+	compile group: 'com.google.code.gson', name: 'gson', version: '2.8.5'
```

Comment:
are these the latest versions available right now? If so, can we mention that above.